src/Authenticator/CookieAuthenticator.php

```diff
@@ -192,8 +192,7 @@ protected function _createToken(ArrayAccess|array $identity): string
 
         $usernameField = $this->getConfig('fields.username');
 
-        /** @var string */
-        return json_encode([$identity[$usernameField], $hash]);
+        return json_encode([$identity[$usernameField], $hash]) ?: '';
```

Comment:
I just (string) cast it usually. But both works.

Reply:
I'd prefer to prevent using the PHP casting system where possible 😅

Reply:
Does it stop complaining about the `false` return if you use the [JSON_THROW_ON_ERROR](https://www.php.net/manual/en/json.constants.php#constant.json-throw-on-error) flag?

We should do that anyway instead of silently returning an empty string.

Reply:
It does stop complaining. thanks